### PR TITLE
Add HomeModifyEvent for home create/delete/rename/set

### PR DESCRIPTION
**Pull request by JRoy**
Att #5213

### CODE REVIEW

File: Essentials/src/main/java/net/essentialsx/api/v2/events/HomeModifyEvent.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+     * returns the updated name if it returns {@link HomeModifyCause#RENAME}, or returns null if it returns {@link HomeModifyCause#DELETE}.
+     * @return The name of the home or null.
+     */
+    public String getNewName() {
+        return newName;
+    }
+
+    /**
+     * Returns the name of the home when {@link #getCause()} returns {@link HomeModifyCause#UPDATE},
+     * returns the previous name if it returns {@link HomeModifyCause#RENAME}, or returns null if it returns {@link HomeModifyCause#DELETE} or {@link HomeModifyCause#CREATE}.
```
Comment (mdcfe) [inline Essentials/src/main/java/net/essentialsx/api/v2/events/HomeModifyEvent.java]: If I understand the constructors right, this should return the name on delete? As described in the doc right now, you would never be able to get the name of a deleted home

Comment (JRoy) [reply Essentials/src/main/java/net/essentialsx/api/v2/events/HomeModifyEvent.java]: Doc is wrong, correct

**Comment by JRoy** [comment]
updated docs

File: Essentials/src/main/java/com/earth2me/essentials/commands/Commanddelhome.java [hunk 2]
```diff
@@ -37,18 +39,24 @@ public void run(final Server server, final CommandSource sender, final String co
...
-        if (name.equalsIgnoreCase("bed")) {
+        if (name.equals("bed")) {
             throw new Exception(tl("invalidHomeName"));
         }
 
-        user.delHome(name.toLowerCase(Locale.ENGLISH));
+        final HomeModifyEvent event = new HomeModifyEvent(sender.getUser(ess), user, name, user.getHome(name), false);
+        Bukkit.getServer().getPluginManager().callEvent(event);
+        if (event.isCancelled()) {
+            return;
```
Comment (mdcfe) [inline Essentials/src/main/java/com/earth2me/essentials/commands/Commanddelhome.java:59]: Similar to #5304, cancellation feedback would make it clearer to users why the home hasn't been modified.

**Review by mdcfe — APPROVED** [review]

**Pull request merged**